pkg/cmd/experimental/sync-groups/types.go

```diff
@@ -0,0 +1,185 @@
+	PreferredUsernameAttribute []string
+	// IDAttribute is the required list of LDAP attributes to use for the id address of the user identity.
+	// The first attribute with a non-empty value is used. If no attributes have values, login fails.
+	IDAttribute []string
+}
+
+// LDAPCommonQueryConfig holds config for an LDAP query that retrieves all relevant entries from the
+// LDAP server - all users, all groups, etc.
+type LDAPCommonQueryConfig struct {
+	// The DN of the branch of the directory where all searches should start from
+	// (e.g. "(OU=OpenShiftDevelopers,OU=Acme,DC=acme,DC=com)")
+	BaseDN string
+	// The scope of the search. Can be ldap.ScopeWholeSubtree, ldap.ScopeSingleLevel, or ldap.ScopeBaseObject
+	Scope ldappassword.Scope
+	// A valid LDAP search filter that retrieves all relevant entries from the LDAP server
+	// (e.g. "(objectClass=User)")
+	CommonFilter string
+}
+
+// LDAPSpecificQueryConfig holds config for a specific LDAP query that retrieves one specific relevant
+// entry from the LDAP server - one user, group, list of members that represent a group, etc.
+type LDAPSpecificQueryConfig struct {
+	// LDAPCommonQueryConfig holds the config for the base query
+	CommonConfig LDAPCommonQueryConfig
+
+	// SpeficFilterBase is the base for the specific filter, to be string-formatted with appropriate
+	// data to form the specific filter that, when conjoined with the common filter, retrieves the
+	// specific LDAP entry from the LDAP server. (e.g. "(&(objectClass=Group)(cn=%s))")
+	SpeficFilterBase string
+}
+
+// LDAPSpecificQueryFromAttributeConfig holds config for a specific LDAP query where the specfic filter
+// is string-formatted with the value of an attribute on a first-class LDAP entry
+type LDAPSpecificQueryFromAttributeConfig struct {
+	// SpecificConfig holds the config for the specific query
+	SpecificConfig LDAPSpecificQueryConfig
+
+	// Attribute is the attribute on the first-class entry whose value will be string-formatted with
+	// the SpecificConfig.SpecificFilterBase to form the specific filter. (e.g. "memberOf")
+	Attribute string
+}
+
+// LDAPServerConfig holds config for a parsed RFC 2255 URL
+type LDAPServerConfig struct {
+	// Scheme is ldap or ldaps
+	Scheme Scheme
+	// Host is the host:port of the LDAP server
+	Host string
+	// Insecure specifies if TLS is required for the connection. If true, either an ldap://... URL or
+	// StartTLS must be supported by the server
+	Insecure bool
+	// TLSConfig holds the TLS options. Only used when Insecure=false
+	TLSConfig *tls.Config
+}
+
+// GroupMeta is a collection of data used to uniquely identify an OpenShift Group and hold LDAP entry
+// attributes that will be transferred to an OpenShift Group during a sync. The GroupFieldAttributeMapping
+// configuration inside of the sync options will define how an data from LDAP first-class group entry is
+// mapped to the fields of a GroupMeta
+type GroupMeta struct {
```

Comment:
This is used in "interfaces.go" move it there.  It's effectively part of your contract.